lib/search.js

```diff
@@ -70,24 +57,50 @@ function search (args, silent, staleness, cb) {
     return s.toLowerCase()
   })
 
-  updateIndex(staleness, opts, searchexclude, filter, function (er, data) {
-    // now data is the list of data that we want to show.
-    // prettify and print it, and then provide the raw
-    // data to the cb.
-    if (er || silent) return cb(er, data)
-    log.clearProgress()
-    output(prettify(data, args))
-    cb(null, data)
+  updateIndex(staleness, function (er, stream) {
+    if (er) return cb(er)
+    log.silly('search', 'searching packages')
+    var matchCount = 0
+    var filterStream = ms.through.obj(function (pkg, enc, cb) {
+      log.gauge.pulse('search')
+      log.gauge.show({section: '🤔 search', logline: 'scanning ' + pkg.name})
```

Comment:
heehee, a+ emoji use buuut... you need to only include it if `npm.config.get('unicode')` is true.